pkg/logger/ecs_error_log.go

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package logger
+
+import (
+	"fmt"
+	"go.uber.org/zap/zapcore"
+	"log"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+)
+
+type ECSErrorLogger struct {
```

Comment:
I think you can paste your PR description here so people would know what it's for when they look at it later on. e.g.

```go
// ECSErrorLogger writes to a file mounted into container from host by ECS Agent.
// The file has a limit of 1MB and only contains error level messages.
// A zap hook is injected to intercept error level log entries 
// if the env var `ECS_ERROR_LOG_FILE_PATH` is valid.
// ....
type ECSErrorLogger struct {
}
```